database/utils.go

```diff
@@ -44,22 +44,22 @@ func SplitOnDupId[T IDer]() com.BulkChunkSplitPolicy[T] {
 	}
 }
 
-// setGaleraOpts sets the "wsrep_sync_wait" variable for each session ensures that causality checks are performed
-// before execution and that each statement is executed on a fully synchronized node. Doing so prevents foreign key
-// violation when inserting into dependent tables on different MariaDB/MySQL nodes. When using MySQL single nodes,
-// the "SET SESSION" command will fail with "Unknown system variable (1193)" and will therefore be silently dropped.
+// unsafeSetSessionVariableIfExists sets the given MySQL/MariaDB system variable for the specified database session.
 //
-// https://mariadb.com/kb/en/galera-cluster-system-variables/#wsrep_sync_wait
-func setGaleraOpts(ctx context.Context, conn driver.Conn, wsrepSyncWait int64) error {
-	galeraOpts := "SET SESSION wsrep_sync_wait=" + strconv.FormatInt(wsrepSyncWait, 10)
-
-	_, err := conn.(driver.ExecerContext).ExecContext(ctx, galeraOpts, nil)
+// NOTE: It is unsafe to use this function with untrusted/user supplied inputs and poses an SQL injection,
+// because it doesn't use a prepared statement, but executes the SQL command directly with the provided inputs.
+//
+// When the "SET SESSION" command fails with "Unknown system variable (1193)", the error will be silently
+// dropped but returns all other database errors.
+func unsafeSetSessionVariableIfExists(ctx context.Context, conn driver.Conn, variable string, value any) error {
+	stmt := fmt.Sprintf("SET SESSION %s=%v", variable, value)
```

Comment:
I would change the type of `value` to `string` to make the behavior of this function a bit more obvious. With `any`, one might expect it to use the [`driver.Valuer`](https://pkg.go.dev/database/sql/driver#Valuer) interface, i.e. that it would work with something like `types.String` which it doesn't. That change also wouldn't make the function harder to use, you'd have to pass `fmt.Sprint(x)` instead of `x` at worst.